doc/manual/metaprogramming.rst

```diff
@@ -134,19 +162,88 @@ ambiguity in parsing.:
     julia> :(::)
     :(::)
 
-:obj:`Symbol`\ s can also be created using :func:`symbol`, which takes
-a character or string as its argument:
+Expressions and evaluation
+--------------------------
+
+Quoting
+~~~~~~~
+
+There is special syntax to create expression objects without using the
+explicit :obj:`Expr` constructor above. A short form for inline expressions
+uses the ``:`` followed by a single parenthesized expression. Here is an
+example of the short form used to quote an arithmetic expression:
 
 .. doctest::
 
-    julia> symbol('\'')
-    :'
+    julia> ex = :(a+b*c+1)
+    :(a + b * c + 1)
 
-    julia> symbol("'")
-    :'
+    julia> typeof(ex)
+    Expr
+
+(to view the structure of this expression, try ``ex.head`` and ``ex.args``,
+or use :func:`dump` as above).
 
-:func:`eval` and Interpolation
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Note that equivalent expressions may be constructed using :func:`parse` or
+the direct :obj:`Expr` form:
+
+.. doctest::
+
+    julia>      :(a + b*c + 1)  ==
+	   parse("a + b*c + 1") ==
+	   Expr(:call, :+, :a, Expr(:call, :*, :b, :c), 1)
+    true
+
+Expressions provided by the parser generally only have symbols, other
+expressions, and literal values as their args, whereas expressions
+constructed by Julia code can have arbitrary run-time values
+without literal forms as args. In this specific example, ``+`` and ``a``
+are symbols, ``*(b,c)`` is a subexpression, and ``1`` is a literal
+64-bit signed integer.
+
+There is a second form of quoting for multiple expressions:
+blocks of code enclosed in ``quote ... end``
+
+.. doctest::
+
+    julia> ex = quote
+		    x = 1
+		    y = 2
+		    x + y
+		end
+    quote  # none, line 2:
+	x = 1 # line 3:
+	y = 2 # line 4:
+	x + y
+    end
+
+    julia> typeof(ex)
+    Expr
+
+Interpolation
+~~~~~~~~~~~~~
+
+Direct construction of :obj:`Expr` objects with value arguments is
+powerful, but :obj:`Expr` constructors can be tedious compared to "normal"
+Julia syntax. Instead, Julia allows "splicing" or interpolation of expression
+objects, prefixed with ``$``, into quoted expressions. The above example
+can be written more clearly and concisely using interpolation:
+
+.. doctest::
+
+    julia> a = 1;
+
+    julia> ex = :($a + b)
+    :(1 + b)
+
```

Comment:
I think this example might be a bit too simple, I feel this is one of the topics people get the most confused by so it might be worth expanding a bit.

Some simple examples I have seen people get tripped up by:
```julia
# wrong
a = 1
ex = :($a in $((1,2,3)))

# right
a = 1
ex = :($a in $(Expr(:tuple, 1,2,3)))
```

I think people get confused because numbers are literal values in the AST so they think that all literal values can be interpolated into the expression.

Another case where this comes up is interpolating symbols which might deserve a special mention.
```julia
# ex
:a in (:a, :b)

#wrong
:($:a in $(Expr(:tuple, :a, :b)))

# right
:($(:(:a)) in $(Expr(:tuple, :(:a), :(:b)))))
```
Something about nested interpolations would also be a good "advanced" example but I cannot think of one off the top of my head.